### ISSUE TITLE
Asks for root password

### ISSUE THREAD

**Issue by TamasBarta** [hyprwm/hyprpolkitagent#6]
I might misunderstand things, but my understanding is that polkit is for privilege escalation for the current user. This agent however doesn't accept my user's password, only the root password. I had root disabled, so it never worked for me, until I set a password for the root user. I'm certain this should not be the case.

**Comment by vaxerski** [comment]
I think that's how it works, do other polkit agents ask for the user password?

**Comment by SaltyBooty** [comment]
> I think that's how it works, do other polkit agents ask for the user password?

From my understanding, if user belongs to the group that is allowed to execute a particular command polkit agent will prompt for user password. If user doesn't belong to a group that is able to execute the command it asks for root password.

**Comment by vaxerski** [comment]
that does not answer my question

**Comment by SaltyBooty** [comment]
> that does not answer my question

other polkit agents do prompt for user password

**Comment by TamasBarta** [comment]
I used the KDE and Gnome agents before, and I was never asked for root password, only for my own. I was always in the ˋwheelˋ group (on Arch), if that makes a difference. Since I don't want my root account to have a password, I went back to the KDE agent for now, and that is asking for my user password.

**Comment by vaxerski** [comment]
odd.

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/user-attachments/files/17624026/patch.txt)
try this?

**Comment by Grub4K** [comment]
Yup that works for me to target the current user: `Authenticating for unix-user:grub4k`

**Comment by kekorder** [comment]
@TamasBarta @Grub4K `identities.at(0);` should get the root user right? My user is in the wheel group so it should be selected except it gets the user with ID 1000 while I am logged in as user with ID 1001. Related to the current issue or not?

**Comment by Grub4K** [comment]
Seems as if the identities list should be filtered for the current user, as identities do no seem sorted in any particular order.

<details><summary><a href="https://github.com/KDE/polkit-kde-agent-1/blob/1e83cdfef577ceb639123e607747bd3cbd3a1d0b/QuickAuthDialog.cpp#L60-L65"><code>Relevant KDE polkit agent code</code></summary>

```c++
    auto idents = qobject_cast<IdentitiesModel *>(m_theDialog->property("identitiesModel").value<QObject *>());
    idents->setIdentities(identities, false);
    if (!identities.isEmpty()) {
        int initialIndex = std::max(0, idents->indexForUser(KUser().loginName()));
        m_theDialog->setProperty("identitiesCurrentIndex", initialIndex);
    }
```

</details>

I am not sure how we could determine if to use the root user or current user ahead of time

**Comment by ghost** [comment]
It would appear as if I'm facing a similar issue.
For me, HPA asks for the password of a seemingly random user (my system has 3: root, me and another user) and I'm always prompted for the password of the 3rd one (most recently created).
I noticed some talk of the [`session.selectedUser = identities.at(0);`](https://github.com/hyprwm/hyprpolkitagent/blob/main/src/core/PolkitListener.cpp#L36) line above which seems to be the issue. I'm puzzled as of why it's returning *that* user, but alas, I don't have much experience in these fields, hopefully my issue brings to light some more context and the issue can be resolved.

Cheers

**Comment by Izumemori** [comment]
I'm also running into this issue. I'm on nixos and have three users, root (id 0), colmena (for remote updates) (id 1001), and my user (id 1000).

Only my user has a password set, yet it always asks for the colmena password. 

Excerpt from the service:
`pam_unix(polkit-1:auth): authentication failure; logname=sam uid=1000 euid=0 tty= ruser=colmena rhost=  user=colmena`

**Comment by pdf** [comment]
Related: #48

**Comment by UncleJ4ck** [comment]
this looks fixed since the qt->hyprtoolkit port in #50. identity selection at `src/core/PolkitListener.cpp:302` is now `preferred = "unix-user:" + geteuid()` and the matching identity becomes `selectedIdx`. previous code defaulted to `identities.at(0)` which is what surfaced the wrong user. @vaxerski safe to close

**Comment by pdf** [comment]
@UncleJ4ck this does not appear to be resolved - either it was never fixed, or it has regressed.

Currently using hyprpolkitagent 0.1.3.

The code looks like it *should* be correct, but I'm still prompted to auth as a different user.

If I replace hyprpolkitagent with the Plasma/KDE agent, the correct user is prompted.
